public_html/wp-content/plugins/wordcamp-coming-soon-page/classes/wccsp-customizer.php

```diff
@@ -178,4 +179,58 @@ public function enqueue_customizer_scripts() {
 			true
 		);
 	}
+
+	/**
+	 * Retrieve the WordCamp status.
+	 *
+	 * @return string
+	 */
+	public function get_status() {
+		$wordcamp_post = get_wordcamp_post();
+
+		if ( isset( $wordcamp_post->ID ) ) {
+			return $wordcamp_post->post_status;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Validate the wccsp_settings[enabled] and maybe prevent disabling the coming soon page.
+	 *
+	 * Non valid field prevents the save, thus also disabling the coming soon page.
+	 *
+	 * @param  boolean $validity Status of the field validity.
+	 * @param  string  $value    Field value.
+	 *
+	 * @return boolean           Status of the field validity.
+	 */
+	public function maybe_prevent_disable( $validity, $value ) {
+		// Short circuit when the value is on.
+		if ( 'on' === $value ) {
+			return $validity;
+		}
+
+		/**
+		 * Short circuit when network admin is changing the value.
+		 *
+		 * On typical WordCamp site, at least some organisers do have administrator role so that
+		 * capability can not be used for this check.
+		 */
+		if ( in_array( 'manage_network', wp_get_current_user()->roles ) ) {
```

Comment:
not sure this is necessary, as a super admin would pass the below condition for deputies too - perhaps we can just write a comment pertaining to this. Also, the return value looks like it should be `$validity`